src/main/kotlin/apply/ui/admin/mail/GroupMailTargetFormDialog.kt

```diff
@@ -0,0 +1,117 @@
+import support.views.createErrorButton
+import support.views.createItemSelect
+import support.views.createPrimaryButton
+
+class GroupMailTargetFormDialog(
+    private val recruitmentService: RecruitmentService,
+    private val evaluationService: EvaluationService,
+    private val mailTargetService: MailTargetService,
+    private val reloadComponent: (List<MailTargetResponse>) -> Unit
+) : Dialog() {
+    private val evaluation = createEvaluationItem()
+    private val recruitment = createRecruitmentItem(evaluation)
+    private val evaluationStatus = createEvaluationStatusItem(evaluation)
+    private val mailTargets: MutableList<MailTargetResponse> = mutableListOf()
+    private val currentMailTargets: VerticalLayout = VerticalLayout(createContent())
+
+    init {
+        width = "800px"
+        height = "90%"
+        add(
+            H2("지원자 정보 조회"),
+            HorizontalLayout(
+                recruitment, evaluation, evaluationStatus,
+                HorizontalLayout(
+                    createPrimaryButton("추가") {
+                        reloadComponent(mailTargets)
+                        close()
+                    },
+                    createErrorButton("취소") {
+                        close()
+                    }
+                ).apply {
+                    justifyContentMode = FlexComponent.JustifyContentMode.END
+                    defaultVerticalComponentAlignment = FlexComponent.Alignment.END
+                },
+            ),
+            currentMailTargets
+        ).apply {
+            setWidthFull()
+        }
+        open()
+    }
+
+    private fun createEvaluationItem(): Select<Evaluation> {
+        return createItemSelect("평가")
+    }
+
+    private fun createRecruitmentItem(evaluation: Select<Evaluation>): Select<RecruitmentResponse> {
+        return createItemSelect<RecruitmentResponse>("모집").apply {
+            setItems(*recruitmentService.findAll().toTypedArray())
+            setItemLabelGenerator { it.title }
+            addValueChangeListener {
+                evaluation.apply {
+                    setItems(*evaluationService.findAllByRecruitmentId(it.value.id).toTypedArray())
+                    setItemLabelGenerator { it.title }
+                }
+            }
+        }
+    }
+
+    private fun createEvaluationStatusItem(
+        evaluation: Select<Evaluation>,
+    ): Select<EvaluationStatus> {
+        return createItemSelect<EvaluationStatus>("모집 상태").apply {
+            setItems(*EvaluationStatus.values())
+            setItemLabelGenerator { it.toText() }
+            addValueChangeListener {
+                val mailTargetResponses = mailTargetService.findMailTargets(evaluation.value.id, it.value)
+                mailTargets.clear()
+                mailTargets.addAll(mailTargetResponses)
+
+                currentMailTargets.apply {
+                    this.removeAll()
+                    this.add(createContent(mailTargetResponses))
+                }
```

Comment:
```suggestion
                currentMailTargetsGrid.apply {
                    setItems(mailTargetResponses)
                }
```
setItem을 사용해 createContent() 메서드를 제거하는 방법도 있을 것 같은데 이런 방법은 어떨까요 ??

Reply:
기존에는 VerticalLayout에 Component(아마 Span 이었던걸로..)를 각각 추가하다 보니 지금처럼 removeAll, add 메서드로 되어있었는데, VerticalLayout으로 감싸지 않고 바로 Grid를 사용한다면 마찌가 제안한대로 변경할 수 있을 것 같아요!